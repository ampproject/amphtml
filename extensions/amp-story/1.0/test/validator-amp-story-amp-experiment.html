<!--
  Copyright 2019 The AMP HTML Authors. All Rights Reserved.
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
      http://www.apache.org/licenses/LICENSE-2.0
  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Passing tests for the amp-experiment component inside amp-story.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-story" src="https://cdn.ampproject.org/v0/amp-story-1.0.js"></script>
  <script async custom-element="amp-experiment" src="https://cdn.ampproject.org/v0/amp-experiment-0.1.js"></script>
  <title>My Story</title>
  <meta name="description" content="Get started with amp-story">
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <link rel="canonical" href="grid-layer-templates.html">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
</head>
<body>
  <amp-story standalone title="Foo" publisher="AMP" publisher-logo-src="http://example.com/foo.png" poster-portrait-src="http://example.com/foo.png">
    <amp-story-page id="fill-template-title">
      <amp-story-grid-layer template="vertical">
        <amp-experiment>
            <script type="application/json">
                {
                  "background-color-test": {
                    "variants": {
                      "yellow": 49,
                      "red": 49
                      }
                    }
                  }
            </script>
        </amp-experiment>
      </amp-story-grid-layer>
    </amp-story-page>
  </amp-story>
</body>
</html>
