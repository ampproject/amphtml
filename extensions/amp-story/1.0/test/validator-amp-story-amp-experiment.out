FAIL
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|    <script async custom-element="amp-story" src="https://cdn.ampproject.org/v0/amp-story-1.0.js"></script>
|    <script async custom-element="amp-experiment" src="https://cdn.ampproject.org/v0/amp-experiment-0.1.js"></script>
|    <title>My Story</title>
|    <meta name="description" content="Get started with amp-story">
|    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|    <link rel="canonical" href="grid-layer-templates.html">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|  </head>
|  <body>
|    <amp-story standalone title="Foo" publisher="AMP" publisher-logo-src="http://example.com/foo.png" poster-portrait-src="http://example.com/foo.png">
|      <amp-story-page id="fill-template-title">
|        <amp-story-grid-layer template="vertical">
|          <amp-experiment>
|              <script type="application/json">
|                  {
|                    "background-color-test": {
|                      "variants": {
|                        "yellow": 49,
|                        "red": 49
|                        }
|                      }
|                    }
|              </script>
|          </amp-experiment>
|
|          <!-- Invalid: bare script not allowed as a child of amp-story-grid-layer -->
|          <script type="application/json">
>>         ^~~~~~~~~
amp-story/1.0/test/validator-amp-story-amp-experiment.html:32:8 Custom JavaScript is not allowed. (see https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml#html-tags) [CUSTOM_JAVASCRIPT_DISALLOWED]
|            {
|              "background-color-test": {
|                "variants": {
|                  "yellow": 49,
|                  "red": 49
|                  }
|                }
|              }
|          </script>
|        </amp-story-grid-layer>
|
|        <!-- Invalid: amp-experiment not allowed as a child of amp-story-page -->
|        <amp-experiment>
>>       ^~~~~~~~~
amp-story/1.0/test/validator-amp-story-amp-experiment.html:45:6 Tag 'amp-experiment' is disallowed as child of tag 'amp-story-page'. Child tag must be one of ['amp-analytics', 'amp-pixel', 'amp-story-cta-layer', 'amp-story-grid-layer', 'amp-story-page-attachment']. (see https://amp.dev/documentation/components/amp-story) [AMP_TAG_PROBLEM]
|            <script type="application/json">
|              {
|                "background-color-test": {
|                  "variants": {
|                    "yellow": 49,
|                    "red": 49
|                    }
|                  }
|                }
|            </script>
|        </amp-experiment>
|
|      </amp-story-page>
|
|      <!-- Invalid: amp-experiment not allowed as a child of amp-story -->
|      <amp-experiment>
>>     ^~~~~~~~~
amp-story/1.0/test/validator-amp-story-amp-experiment.html:61:4 Tag 'amp-experiment' is disallowed as child of tag 'amp-story'. Child tag must be one of ['amp-analytics', 'amp-consent', 'amp-geo', 'amp-pixel', 'amp-sidebar', 'amp-story-access', 'amp-story-auto-ads', 'amp-story-bookend', 'amp-story-page']. (see https://amp.dev/documentation/components/amp-story) [AMP_TAG_PROBLEM]
|          <script type="application/json">
|            {
|              "background-color-test": {
|                "variants": {
|                  "yellow": 49,
|                  "red": 49
|                  }
|                }
|              }
|          </script>
|      </amp-experiment>
|    </amp-story>
|  </body>
|  </html>
